### PR TITLE
MODINVSTOR-927: InstallUpgradeIT

### PR DESCRIPTION
**Pull request by julianladisch**
Add an InstallUpgradeIT.java integration test. It should check the shaded fat uber jar and Dockerfile:

* Test /admin/health.
* Test that logging works.
* Test installation and migration with smoke test.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -18,8 +18,7 @@ buildMvn {
     buildJavaDocker {
       publishPreview = false
       publishMaster = 'yes'
-      healthChk = 'yes'
-      healthChkCmd = 'curl -sS --fail -o /dev/null  http://localhost:8081/apidocs/ || exit 1'
+      // healthChk has been moved into InstallUpgradeIT.java
```
Comment (marcjohnson-kint) [inline Jenkinsfile:21]: What does this mean?

As I understand it, this configuration is about how the docker container build is configured, how is an integration test relevant to that configuration?

Comment (steveellis) [reply Jenkinsfile:21]: @julianladisch I share Mark's question, but I think that removing this from the Jenkinsfile means is that docker is no longer responsible for testing the health check endpoint. Instead you have put it into the new test class which calls it using testcontainers. It also does two other things (logging test, and data migration test) which seem like great things to test this way.

How many other modules use this pattern for smoke type tests? Is this something we should consider implementing more widely?

**Review by steveellis — APPROVED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (julianladisch) [reply Jenkinsfile:21]: healthChkCmd is a workaround for modules that doesn't check whether the Docker image actually works in the maven integration test. healthChkCmd is run in the Jenkins build pipeline. It checks whether the the Dockerfile created in the previous step of the build pipeline actually works. This check is performed by running healthChkCmd.

This pull request adds the integration test InstallUpgradeIT that exactly does this check: It creates the Docker image and executes it. This is clearly documented in the javadoc of InstallUpgradeIT:

> Check the shaded fat uber jar and Dockerfile

Therefore the healthChkCmd is no longer needed, it should be removed to save build pipeline resources.

Comment (marcjohnson-kint) [reply Jenkinsfile:21]: > healthChkCmd is a workaround for modules that doesn't check whether the Docker image actually works in the maven integration test. healthChkCmd is run in the Jenkins build pipeline. It checks whether the the Dockerfile created in the previous step of the build pipeline actually works. This check is performed by running healthChkCmd.

I wasn't aware that this is a workaround.

> Therefore the healthChkCmd is no longer needed, it should be removed to save build pipeline resources.

Should all modules move to this approach?

Is this documented somewhere?

**Pull request merged**